parameter/criterion/src/InclusiveCriterion.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "criterion/InclusiveCriterion.h"
+#include "Tokenizer.h"
+
+#include <sstream>
+#include <cassert>
+#include <algorithm>
+
+namespace core
+{
+namespace selection
+{
+namespace criterion
+{
+
+const std::string InclusiveCriterion::gDelimiter = "|";
+
+InclusiveCriterion::InclusiveCriterion(const std::string& name,
+                                       const Values& values,
+                                       core::log::Logger& logger)
+    : Criterion(name, logger, values, {},
+                {{"Includes", [&](const State& state) {
+                    return std::includes(mState.begin(), mState.end(),
+                                         state.begin(), state.end()); }},
+                 {"Excludes", [&](const State& state) {
+                    State inter;
+                    std::set_intersection(mState.begin(), mState.end(),
```

Comment:
I'm saying this because rules can only be

    SomeCriterionName Excludes SomeValue

i.e., it can't be `SomeCriterionName Excludes {SomeValueA, SomeValueB}`.